### PR TITLE
(feat) O3-2798: Add metrics extension slot to the home page

### PR DESCRIPTION
**Pull request by arodidev**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
This PR introduces an extension slot that will cater for the addition of metrics widgets to the home page of the Ref App

## Screenshots

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Review by CynthiaKamau — APPROVED** [review]

File: packages/esm-home-app/src/homepage-metrics/home-metrics-widgets.component.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { ExtensionSlot } from '@openmrs/esm-framework';
+import React from 'react';
+import styles from './home-metrics-widgets.scss';
+
+interface HomeMetricsWidgetProps {}
+
+const HomeMetricsWidgets: React.FC<HomeMetricsWidgetProps> = () => {
+  return <ExtensionSlot className={styles.homeMetricsWidget} name="home-metrics-widgets-slot" />;
```
Comment (ibacher) [inline packages/esm-home-app/src/homepage-metrics/home-metrics-widgets.component.tsx:8]: I don't think we need the class name if there's no style applied yet.

File: packages/esm-home-app/src/index.ts [hunk 2]
```diff
@@ -24,6 +25,8 @@ export const homeWidgetDbLink = getSyncLifecycle(createDashboardLink(dashboardMe
 
 export const homeWidgetDashboard = getSyncLifecycle(homeWidgetDashboardComponent, options);
 
+export const homeMetricsWidgetDashboard = getSyncLifecycle(HomeMetricsWidgetsComponent, options);
```
Comment (ibacher) [inline packages/esm-home-app/src/index.ts:28]: Please use `getAsyncLifecycle()`. Basically, we should _always_ use `getAsyncLifecycle()` as it improves app performance.

**Review by ibacher — DISMISSED** [review]
Why are we adding an extension to render a slot on the home page? I'd just put the `ExtensionSlot` component on the home page itself.

File: packages/esm-home-app/src/index.ts (hunk 2 above)
Comment (arodidev) [reply packages/esm-home-app/src/index.ts:28]: Noted, thanks.

**Review by denniskigen — APPROVED** [review]
Thanks, @arodidev.

**Comment by arodidev** [comment]
This PR introduces a bug where the extension slot is not placed correctly within the UI and is corrected in [this PR](https://github.com/openmrs/openmrs-esm-home/pull/191)

**Pull request merged**